parley/src/layout/alignment.rs

```diff
@@ -7,31 +7,84 @@ use super::{
 };
 use crate::style::Brush;
 
+/// Additional options to fine tune alignment
+#[derive(Debug, Clone)]
+pub struct AlignmentOptions {
+    /// If set to `true`, "end" and "center" alignment will apply even if the line contents are
+    /// wider than the alignment width. If it is set to `false`, all overflowing lines will be
+    /// [`Alignment::Start`] aligned.
+    pub align_when_overflowing: bool,
```

Comment:
A nit: it may read more naturally by flipping the meaning of the bool and calling the field `start_alignment_on_overflow`.